### PR TITLE
Update user reducer

### PR DESCRIPTION
**Pull request by stufreen**
Changes:
* Move the `self` and `auth` data to the `user` reducer
* Add some networks requests in `app/services/network.js` relating to sign in, sign out, forgot password, change password, and change name
* Remove the API middleware and `core` reducer
* Improve Login, Sign up, Forgot password, and Settings forms. In each case we leverage Redux Form more for front end validation and use Redux Form's submit validation for a better user experience.
* Delete files after they are uploaded or when the user logs out
* Display a warning popup if the user tries to log out when they have uploads in progress

### CODE REVIEW

File: android/app/build.gradle [hunk 1]
```diff
@@ -101,8 +101,8 @@ android {
         applicationId "lab.childmindinstitute.data"
         minSdkVersion rootProject.ext.minSdkVersion
         targetSdkVersion rootProject.ext.targetSdkVersion
-        versionCode 14
-        versionName "0.1.4"
+        versionCode 5
```
Comment (shnizzedy) [inline android/app/build.gradle]: `versionCode 7`

File: ios/MDCApp/Info.plist [hunk 2]
```diff
@@ -17,11 +17,11 @@
 	<key>CFBundlePackageType</key>
 	<string>APPL</string>
 	<key>CFBundleShortVersionString</key>
-	<string>0.1.4</string>
+	<string>0.1.5</string>
 	<key>CFBundleSignature</key>
 	<string>????</string>
 	<key>CFBundleVersion</key>
-	<string>14</string>
+	<string>5</string>
```
Comment (shnizzedy) [inline ios/MDCApp/Info.plist]: `<string>7</string>`

**Comment by stufreen** [comment]
@shnizzedy Changes made. I updated the version to `0.1.7` and the version codes to `7`, is that right?

**Comment by shnizzedy** [comment]
> I updated the version to 0.1.7 and the version codes to 7, is that right?

I meant 0.1.5(7), but as long as we didn't have an active 0.1.7(7) on TestFlight (we didn't), this should work just fine. Thanks!

**Pull request merged**